firmware/application/ui_record_view.cpp

```diff
@@ -113,7 +113,7 @@ uint32_t RecordView::set_sampling_rate(uint32_t new_sampling_rate) {
      * They are ok as recorded spectrum indication, but they should not be used by Replay app. (the voice speed will be accelerated)
 
      * We keep original black background in all the correct IQ .C16 files BW's Options. */
-    if ((actual_sampling_rate > 8'000'000) || (actual_sampling_rate <= 1'600'000)) {  // yellow REC button means not ok for REC, BW >1Mhz , BW <= 100khz due to NG aliasing.
+    if ((actual_sampling_rate > 8'000'000) || ((actual_sampling_rate <= 1'600'000) && (oversample_rate) > OversampleRate::x8)) {  // yellow REC button means not ok for REC, BW >1Mhz , BW <= 100khz due to NG aliasing.
```

Comment:
As discussed via Discord, I would suggest moving the ")" after oversample_rate to the end of the line, for clarity.